### DISCUSSION TITLE
implementation: data fetchers and subscriptions

### DISCUSSION THREAD

**Discussion by xenoterracide** [Netflix/dgs-framework#211] [Q&A]
Apollo Server has a problem with its implementation of subscriptions and data loaders. In Apollo Server your data loaders are loaded in the context, to be reloaded on every request, problem is that a subscription is only a single request, which meant the data loader never uncached. I found a workaround for that problem.

How did DGS avoid this problem in its implementation?

**Comment by paulbakker** [comment]
I’m not familiar with the problem in Apollo Server, so it’s hard to give a useful answer to this. The code for (websocket) subscriptions is here if you’re interested: https://github.com/Netflix/dgs-framework/blob/master/graphql-dgs-subscriptions-websockets/src/main/kotlin/com/netflix/graphql/dgs/subscriptions/websockets/DgsWebSocketHandler.kt
